onnx/defs/parser.cc

```diff
@@ -86,8 +86,71 @@ Status OnnxParser::Parse(TypeProto& typeProto) {
       // Create shape with zero dimensions for scalar
       (void)(tensortype->mutable_shape());
     }
-  } else
-    return ParseError("Unexpected type.");
+  } else {
+    switch (KeyWordMap::Lookup(id)) {
+      case KeyWordMap::KeyWord::SEQ_TYPE: {
+        // Grammar: seq ( type )
+        MATCH('(');
+        auto* seqtype = typeProto.mutable_sequence_type();
+        PARSE(*seqtype->mutable_elem_type());
```

Comment:
Does this support nested types such as `seq(seq(float))`? Maybe throw if we don't want nested types.

Reply:
We want to allow such nested types, and the parser allows it. It is up to operators to restrict allowed types.